src/components/API.js

```diff
@@ -0,0 +1,85 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleRes(res) {
```

Comment:
It's great to have a function for processing the server response, so the logic is not duplicated